src/tests/ftest/cart/util/cart_utils.py

```diff
@@ -158,12 +158,10 @@ def cleanup_processes(self):
         negative_filter = r"'\<(grep|defunct)\>'"
         retry_count = 0
         while retry_count < 2:
-            if stop_processes(self.log, localhost, processes, negative_filter=negative_filter):
-                self.log.info("Stopped '%s' processes on %s", processes, localhost)
-                retry_count += 1
-            else:
-                self.log.info("All '%s' processes have been stopped", processes)
-                retry_count = 99
+            _, running = stop_processes(self.log, localhost, processes, exclude=negative_filter)
+            if not running:
+                retry_count = 2
+            retry_count += 1
         if retry_count == 2:
             error_list.append("Unable to stop cart processes!")
```

Comment:
Pre-existing, but I think the usage of `retry_count` is confusing here. Personally, I would do something like:
```
        running = True
        for _ in range(2):
            _, running = stop_processes(self.log, localhost, processes, exclude=negative_filter)
            if not running:
                break
        if running
            error_list.append("Unable to stop cart processes!")
```